charts/nginx-gateway-fabric/templates/scc.yaml

```
@@ -0,0 +1,40 @@
+{{- if and (eq .Values.platform "openshift") .Values.createOpenShiftSCC }}
+kind: SecurityContextConstraints
+apiVersion: security.openshift.io/v1
+metadata:
+  name: {{ include "nginx-gateway.scc-name" . }}
+allowPrivilegeEscalation: {{ .Values.nginxGateway.securityContext.allowPrivilegeEscalation }}
+allowHostDirVolumePlugin: false
+allowHostIPC: false
+allowHostNetwork: false
+allowHostPID: false
+allowHostPorts: false
+allowPrivilegedContainer: false
+readOnlyRootFilesystem: true
+runAsUser:
+  type: MustRunAsRange
+  uidRangeMin: 101
+  uidRangeMax: 102
+fsGroup:
+  type: MustRunAs
+  ranges:
+  - min: 1001
+    max: 1001
+supplementalGroups:
+  type: MustRunAs
+  ranges:
+  - min: 1001
+    max: 1001
+seLinuxContext:
+  type: MustRunAs
+volumes:
+- emptyDir
+- secret
+users:
+- {{ printf "system:serviceaccount:*:%s" (include "nginx-gateway.fullname" .) }}
```

why do we need * in `"system:serviceaccount:*:`?

---

That is for the namespace/project name, so the * is the wildcard in the case they deploy NGF to a different namespace/project. 

https://github.com/nginxinc/nginx-ingress-helm-operator/blob/main/resources/scc.yaml NIC does the same too, so I just kept the logic.

---

We can change the wildcard to the Helm release namespace - as per the other conversations, this SCC will have a 1-2-1 mapping to the ServiceAccount. And actually, the ServiceAccountName can be overridden, so I think this should be:

```suggestion
- {{ printf "system:serviceaccount:%s:%s" .Release.Namespace (include "nginx-gateway.serviceAccountName" .) }}
```

---

Ok, this seems to work, so I'll add it in, but why do you mean by the SCC will have a 1-2-1 mapping?

---

For NIC, the SCC is created once in the cluster and reused by all instantiations of NIC (because it uses an operator model). This means that it needs to work across multiple namespaces. Here, any installations of NGF will have it's own dedicated SCC, so it can, and probably should, be namespace scoped.